### DISCUSSION TITLE
Unable to Edit Live Course Due to Permission Error

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-Ed/Community#753] [General Discussion]
Hello Team,

I am facing an issue while trying to edit a live course, where I get the error: "You don't have permission to edit the course."

The API used to fetch the course content details returns a status of "Failed."

**API Details:

API URL:**
`https://sunbirdsaas.com/api/content/v1/read/do_114110034185838592192?fields=createdBy,status,mimeType,contentType,resourceType,collaborators,contentDisposition,primaryCategory,framework,targetFWIds&mode=edit`

**Response:**
`{
   "responseCode": "OK",
   "result": {
       "content": {
           "mimeType": "application/vnd.ekstep.content-collection",
           "objectType": "Content",
           "primaryCategory": "Course",
           "contentType": "Course",
           "identifier": "do_114110034185838592192",
           "languageCode": ["en"],
           "status": "Failed",
           "contentDisposition": "inline",
           "targetFWIds": ["education"],
           "framework": "education",
           "resourceType": "Course"
       }
   }
}`

Additionally, when searching for this content using the search API with status=failed, the following response is received:
`{
    "identifier": "do_114110034185838592192",
    "lastPublishedOn": "2024-08-01T07:31:56.124+0000",
    "publishError": "key not found: do_114110034760663040193",
    "name": "Grade 7",
    "createdOn": "2024-07-31T08:58:02.457+0000",
    "objectType": "Content",
    "status": "Failed"
}`

When trying to read the resource do_114110034760663040193, I get a count of 0, and I’m unable to search for this content. Moreover, this resource does not appear in “All My Content,” but it shows up during a global search.

**Troubleshooting Steps Tried:**
Content Publishing issue due to key not found error.
1. Need to open neo4j and also cassandra
2. cqlsh
3. DESC Keysoaces
4. DESC prod_hierachy_store; you will see, conent_heirrchy, freamewokr_hierarchy, questionset_hierarchy
5. DESC prod_hierarchy_storey.content_heirarchy
6. select relational_metaata from prod_hierarchy_storey.content_heirarchy WHERE identifier='333.img'
7.Go to neo4j  match (n:domain) where n.IL_UNIQUE_iD="do_xx" return n. relational_metadata
8. match (n:domain) where n.IL_UNIQUE_iD="do_xx.img" set n. relational_metadata = null
Update in Cassandra also
update prod_hierarchy_store.content_heirarchy set relational_metadta=null WHERE identifier='do_ddd.img'
Then publish by going to Learning server

**Request:**
Please assist with resolving the "key not found" error and permission issue so that I can edit and publish the course successfully.

cc: @HarishGangula

**Comment by HarishGangula** [comment]
@siddhishinde0723 This issue belongs to knowlg building block please raise this issue in https://github.com/orgs/Sunbird-Knowlg/discussions
